File: 26-remove-duplicates-from-a-sorted-array.js
var removeDuplicates = function(nums) {
    let unique=[]
    for(let i = 0; i<nums.length;i++){
        if(!unique.includes(nums[i])){
            unique.push(nums[i])
        }
    }
    for (let i = 0; i < unique.length; i++) {
        nums[i] = unique[i];
    }
    return unique.length
};

File: 26-remove-duplicates-from-a-sorted-array.py
#logically correct but not fit for this problem 
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        expectedNums=[]
        for i in range(len(nums)):
            if nums[i] not in expectedNums:
                expectedNums.append(nums[i])
            nums[:]=expectedNums
        return len(nums)
        
#better approach
class Solution: 
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums:
            return 0
        start=0
        for i in range (1,len(nums)):
            if nums[i]!=nums[start-1]:
                nums[start]=nums[i]
                start+=1
        return start+1   
    
